### DISCUSSION TITLE
Could we omit eager mode on paste?

### DISCUSSION THREAD

**Discussion by ekhodx** [uNmAnNeR/imaskjs#1010] [Ideas]
Example: mask `https://f\\acebook.com/x` with x (`/[0-9]|[A-Za-z]|_/`) repeated infinitely

When I enter value manually, I would like to not see `https://faceb` when I have pressed "b", this is where eager mode is helpful. But enabling it will break UX when user would like to paste full link, which would result in `https://facebook.com/https://www.facebook.com/group123`

Maybe an option like `disableEagerOnPaste: boolean`?

@uNmAnNeR have you ever experienced an UX issue like this in apps you use or develop? Is this doable with imaskjs?

**Comment by isqua** [comment]
I experience the same issue with mask for URLs ([run example in CodeSandbox](https://codesandbox.io/p/sandbox/imask-discussions-1010-sdmmn6?file=%2Fsrc%2Findex.mjs%3A25%2C1)):

```javascript
{
  placeholder: "https://",
  mask: "https://pathname",
  blocks: {
    pathname: {
      mask: /./,
    }
  }
}
```

But it is not actually depends on pasting. The mask behaves the same when typing. The problem is, with `eager: true` it doubles the matching part, but with no `eager` it fills the part with any matching sequence:

— | eager: false | eager: true
-- | -- | --
Type manually `test.com` | ⛔️ Result: `https://est.com` | ✅ Result: `https://test.com`
Paste `test.com` | ⛔️ Result: `https://est.com` | ✅ Result: `https://test.com`
Type manually `https://test.com` | ✅ Result: `https://test.com` | ⛔️ Result: `https://https://test.com`
Paste `https://test.com` | ✅ Result: `https://test.com` | ⛔️ Result: `https://https://test.com`

I also have an assumption that the ` symbol can help here in some way, but from the documentation I did not understand exactly how it works and when to use it.

**Comment by isqua** [comment]
@uNmAnNeR, thank you for the library!

Could you please add an example or some docs about this case?

**Reply by ekhodx** [reply]
@isqua maybe you can try to convert my discussion into an issue, as it gets no attention at all
